internal/bootstrap/agentbinary.go

```diff
@@ -33,9 +34,21 @@ type AgentBinaryStorage interface {
 	Add(context.Context, io.Reader, binarystorage.Metadata) error
 }
 
+// AgentBinaryStore is the service used to persist Juju agent binaries into the
+// controller.
+type AgentBinaryStore interface {
+	AddWithSHA256(context.Context, io.Reader, coreagentbinary.Version, int64, string) error
+}
+
 // PopulateAgentBinary is the function that is used to populate the agent
 // binary at bootstrap.
-func PopulateAgentBinary(ctx context.Context, dataDir string, storage AgentBinaryStorage, logger logger.Logger) (func(), error) {
+func PopulateAgentBinary(
+	ctx context.Context,
+	dataDir string,
+	storage AgentBinaryStorage,
```

Comment:
Maybe add a TODO to mention that this is the mongo store and we should not forget to remove it later.

Reply:
Done through Ian's feedback.